src/etc/inc/certs.inc

```diff
@@ -269,6 +269,48 @@ function cert_create(&$cert, $caref, $keylen, $lifetime, $dn, $digest_alg = 'sha
     return true;
 }
 
+function sign_cert_for_csr(&$cert, $caref, $csr, $lifetime, $digest_alg, $x509_extensions = 'usr_cert')
```

Comment:
Please remove 'for_' in the function name.

Reply:
Also,  $x509_extensions is limited to 'usr_cert' which is bad.

1. It drops any X509v3 subjectAltName attribute values supplied in the CSR while they should indeed be copied 1:1

2. It drops any X509v3 Key Usage attributes while they should indeed be copied 1:1

3. It also adds a meaningless comment: `nsComment = "OpenSSL Generated Certificate"`

Example X509v3 extensions from a CSR I just signed using the code in here:

- CSR
```
        Requested Extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            X509v3 Key Usage: 
                Digital Signature, Non Repudiation, Key Encipherment
            X509v3 Subject Alternative Name: 
                DNS:my.fqdn.tld
```
- Signed certificate:
```
        X509v3 extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            Netscape Comment: 
                OpenSSL Generated Certificate
            X509v3 Subject Key Identifier: 
                38:18:41:5C:9C:07:68:6F:DF:99:63:A2:3E:8E:42:3D:94:AB:3D:41
            X509v3 Authority Key Identifier: 
                keyid:38:01:5E:58:A1:1D:C2:4D:E6:62:03:8D:E1:EF:42:4F:57:BE:C2:94
```